core/local/steps/buffer.js

```diff
@@ -23,6 +23,8 @@ module.exports = class Buffer {
   }
 
   push (batch /*: Batch */) /*: void */ {
+    if (batch.length === 0) return
```

Comment:
The commit description should actually be something like `core/local/steps/buffer: Drop empty batches`.
Independant of the default watcher change.
Except for this, looks good :+1:

Reply:
https://github.com/cozy-labs/cozy-desktop/pull/1472